File: 1. Creational Design Patterns/Singleton/Java/SingletonExample.java
// Singleton Pattern Example in Java
// Thread-safe initialization via static holder idiom

class AppConfig {
    private AppConfig() {}
    private String connectionString = "jdbc:mysql://localhost:3306/app";

    private static class Holder {
        private static final AppConfig INSTANCE = new AppConfig();
    }

    public static AppConfig getInstance() {
        return Holder.INSTANCE;
    }

    public void print() {
        System.out.println("Connection: " + connectionString);
    }
}

public class SingletonExample {
    public static void main(String[] args) {
        AppConfig a = AppConfig.getInstance();
        AppConfig b = AppConfig.getInstance();
        System.out.println(a == b); // true
        a.print();
    }
}

File: 1. Creational Design Patterns/Singleton/Kotlin/SingletonExample.kt
// Singleton Pattern Example in Kotlin
// Uses Kotlin's object declaration which is thread-safe and lazy by default.

object AppConfig {
    val connectionString: String = "jdbc:sqlite:app.db"
    fun printConn() = println("Connection: $connectionString")
}

fun main() {
    val a = AppConfig
    val b = AppConfig
    println(a === b) // true
    a.printConn()
}
